web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/RekoStatusDecoratorImpl.java

```diff
@@ -69,7 +69,8 @@ private RekoStatusDTO getRekoStatus(List<Reko> rekoStatuses,
         final var rekoStatusFromDb = rekoStatuses
                 .stream()
                 .filter((status) -> status.getPatientId().equals(patientId)
-                        && status.getSickLeaveTimestamp().isAfter(sickLeave.getStart().atStartOfDay())
+                        && (status.getSickLeaveTimestamp().isAfter(sickLeave.getStart().atStartOfDay())
```

Comment:
Nice finding, this should fix our problem with reko not beeing updated correctly! Great that u added tests to verify the behaviour!

I think we can make this stream alittle easier to understand by breaking it out into describing functions, maybe something like this?

```java
    private RekoStatusDTO getRekoStatus(List<Reko> rekoStatuses,
        String patientId,
        SjukfallEnhet sickLeave) {
        final var rekoStatusFromDb = rekoStatuses
            .stream()
            .filter(status -> status.getPatientId().equals(patientId))
            .filter(status -> equalsOrAfterStartDate(sickLeave, status))
            .filter(status -> beforeEndDate(sickLeave, status)
            )
            .max(Comparator.comparing(Reko::getRegistrationTimestamp));

    private boolean beforeEndDate(SjukfallEnhet sickLeave, Reko status) {
        return status.getSickLeaveTimestamp().isBefore(sickLeave.getSlut().plusDays(1).atStartOfDay());
    }

    private static boolean equalsOrAfterStartDate(SjukfallEnhet sickLeave, Reko status) {
        return status.getSickLeaveTimestamp().isAfter(sickLeave.getStart().atStartOfDay())
            || sickLeave.getStart().atStartOfDay().equals(status.getSickLeaveTimestamp());
    }

Reply:
Good suggestion